cmd/controller-manager/app/controller_manager.go

```diff
@@ -90,19 +90,20 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 	// Build the K8s kubeconfig / client / clientBuilder
 	glog.V(4).Info("Building k8s kubeconfig")
 
-	k8sKubeconfig, err := rest.InClusterConfig()
+	var err error
+	var k8sKubeconfig *rest.Config
+	if controllerManagerOptions.K8sAPIServerURL == "" && controllerManagerOptions.K8sKubeconfigPath == "" {
```

Comment:
yeah, mainly.  Plus, BuildConfigFromFlags always logs a message that's confusing at best, since it mentions flags that are not the same names as the ones we use.